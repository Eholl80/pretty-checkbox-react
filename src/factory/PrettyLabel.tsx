import * as React from 'react';

import classNames from 'classnames';
import { CommonCheckboxRadioProps } from '../types/CommonProps';

export type PrettyLabelProps = Pick<CommonCheckboxRadioProps, 'color' | 'children' | 'baseId'> & {
    icon?: any;
};

export const PrettyLabel: React.FC<PrettyLabelProps> = ({
    color,
    icon,
    children,
    baseId,
}: PrettyLabelProps) => (
    <div
        className={classNames('state', {
            [`p-${color}`]: color,
        })}>
        {icon}
        <label htmlFor={baseId}>{children}</label>
    </div>
);

PrettyLabel.displayName = 'Pretty.Label';
