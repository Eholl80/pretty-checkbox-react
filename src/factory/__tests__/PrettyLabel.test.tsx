import * as React from 'react';
import { render } from '@testing-library/react';
import { PrettyLabel } from '../PrettyLabel';

describe('PrettyLabel tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<PrettyLabel />).container;
        }).not.toThrow();
    });

    it('should render children inside the label', () => {
        const { getByText, container } = render(<PrettyLabel baseId="abc">Testing</PrettyLabel>);
        getByText('Testing');
        expect(container).toMatchSnapshot();
    });

    it('should append colors to the root div', () => {
        expect(
            render(<PrettyLabel color="danger">Testing</PrettyLabel>).container
        ).toMatchSnapshot();
    });
});
