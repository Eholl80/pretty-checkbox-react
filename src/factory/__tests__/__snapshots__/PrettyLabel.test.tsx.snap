// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrettyLabel tests should append colors to the root div 1`] = `
<div>
  <div
    class="state p-danger"
  >
    <label>
      Testing
    </label>
  </div>
</div>
`;

exports[`PrettyLabel tests should render children inside the label 1`] = `
<div>
  <div
    class="state"
  >
    <label
      for="abc"
    >
      Testing
    </label>
  </div>
</div>
`;
